### ISSUE TITLE
Test case arguments are not displayed in output

### ISSUE THREAD

**Issue by Nysosis** [alsatian-test/alsatian#308]
I **think** it's because of the use of the reserved word `arguments` as the name of the property on [ITestCase](https://github.com/alsatian-test/alsatian/blob/master/core/_interfaces/test-case.i.ts), locally I've changed it to `caseArguments` and started seeing expected behaviour.

Node v6.9.1
Typescript v2.1.6
Windows 7

This happens on the tap and non-tap output:

```
DummyTests
Add
|====================|

Pass: 1/1
Fail: 0/1
Ignore: 0/1
```

```
TAP version 13
1..1
# FIXTURE DummyTests
ok 1 Add
```

After changing to use of `caseArguments`:

```
DummyTests
Add [ 1, 1, 2 ]
|====================|

Pass: 1/1
Fail: 0/1
Ignore: 0/1
```

```
TAP version 13
1..1
# FIXTURE DummyTests
ok 1 Add [ 1, 1, 2 ]
```

**Comment by jamesrichford** [comment]
Great spot, which version of Alsatian are you using?

**Comment by Nysosis** [comment]
v1.2.0, I'm on my fork, but that's at the same point the main repo is at.

**Comment by jamesrichford** [comment]
Cool, we'll just need to check 1.1 and 1.0 for applying this fix too - should be quite simple as you've said :)

**Comment by Nysosis** [comment]
I've committed the fix to my branch, but just wondering, would it be better to enclose the arguments in curly brackets `Add ( 1, 1, 2 )`, as opposed to square `Add [ 1, 1, 2 ]`?
That way it looks more like a method call, and is clearer that the test case wasn't called with one argument which was an array of values, (but that's just my opinion).

**Comment by jamesrichford** [comment]
Yep that looks like a good idea to me 👍 I've had a backwards check and it looks like this affects 1.1.0 but not 1.0.0 or 1.0.1 is this what you're finding?

I'm also putting in a task to run CI for Windows too as this task has reminded me that we're missing it :)

**Comment by Nysosis** [comment]
Yup, I'm seeing the same, 1.1.0 is not showing them, 1.0.1 is working as expected.

I'll take a look at tweaking the squares to curlys tomorrow morn.

How do you want to manage patching up 1.1.0?

**Comment by jamesrichford** [comment]
Perfect, nice one.

You should be able to send a separate PR to the 1.1.x branch which will enable us to release this as a patch. Might need to do the same for 1.2.x as it's already branched

**Comment by Jameskmonger** [comment]
@jamesrichford Should we really be fixing previous branches? It seems like the best thing to do would be to just fix it in the latest version and release that rather than having three separate development versions.

**Comment by jamesrichford** [comment]
@Jameskmonger so obviously we need it in master so that's the reason for that being there

To get it into 1.2.1 we'll need to put it into the 1.2.x branch. The reason this branch exists now is so that development work towards 1.3.0 can go straight into master without causing issues for any patches for the current release.

The reason for the third one is that considering the issue it's probably worth the minimal extra effort to path 1.1.0 so that this feature works again as it's quite an obvious feature and certainly one that is a differentiation vs other test frameworks so good to highlight.

All make sense? :)

**Comment by Jameskmonger** [comment]
@jamesrichford I just think that given that 1.2.0 is now released on npm, we shouldn't be modifying any minor releases before that - anyone experiencing an issue with 1.1.0 should update to 1.2.0, as there are no breaking changes.

As a side note, I'm fairly sure that npm won't allow us to publish any 1.1.x now that 1.2.0 has been published.

**Comment by jamesrichford** [comment]
OK, so I guess my point is that by patching this previous version we're giving people the choice to make a patch upgrade or a minor upgrade. I think for lower impact issues that just patching the latest would be fine but it's just a nice addition I think considering it doesn't take me too much effort to perform the patch.

What were your reasons for thinking not to patch a previous version as a rule?

**Comment by Jameskmonger** [comment]
@jamesrichford I think that patching previous major versions is fine as people may not want to update to the latest major and risk breaking changes. With a minor version though there should be no breaking changes, so I can't see a reason for someone to upgrade from `1.1.2` to `1.1.3` to fix a bug if `1.2.1` is released with that bug fix. We should discourage people from staying on out-of-date minor versions by releasing updates to the latest minor only.

**Comment by jamesrichford** [comment]
Cool, lets go with 1.2 only for now. I wanna keep people on the latest if possible
